lib/if.c

```diff
@@ -1092,6 +1093,73 @@ const char *if_link_type_str(enum zebra_link_type llt)
 	return NULL;
 }
 
+bool if_link_params_cmp(struct if_link_params *iflp1,
+			struct if_link_params *iflp2)
+{
+	if (iflp1->lp_status != iflp2->lp_status)
+		return false;
+	if (iflp1->te_metric != iflp2->te_metric)
+		return false;
+	if (iflp1->default_bw != iflp2->default_bw)
+		return false;
+	if (iflp1->max_bw != iflp2->max_bw)
+		return false;
+	if (iflp1->max_rsv_bw != iflp2->max_rsv_bw)
+		return false;
+	if (iflp1->admin_grp != iflp2->admin_grp)
+		return false;
+	for (int i = 0; i < MAX_CLASS_TYPE; i++) {
+		if (iflp1->unrsv_bw[i] != iflp2->unrsv_bw[i])
+			return false;
+	}
+	if (!admin_group_cmp(&iflp1->ext_admin_grp, &iflp2->ext_admin_grp))
+		return false;
+	if (iflp1->rmt_as != iflp2->rmt_as)
+		return false;
+	if (memcmp(&iflp1->rmt_ip, &iflp2->rmt_ip, sizeof(iflp1->rmt_ip)))
+		return false;
+	if (iflp1->av_delay != iflp2->av_delay)
+		return false;
+	if (iflp1->min_delay != iflp2->min_delay)
+		return false;
+	if (iflp1->max_delay != iflp2->max_delay)
+		return false;
+	if (iflp1->delay_var != iflp2->delay_var)
+		return false;
+	if (iflp1->pkt_loss != iflp2->pkt_loss)
+		return false;
+	if (iflp1->res_bw != iflp2->res_bw)
+		return false;
+	if (iflp1->ava_bw != iflp2->ava_bw)
+		return false;
+	if (iflp1->use_bw != iflp2->use_bw)
+		return false;
+	return true;
+}
+
+void if_link_params_copy(struct if_link_params *dst, struct if_link_params *src)
+{
+	dst->lp_status = src->lp_status;
+	dst->te_metric = src->te_metric;
+	dst->default_bw = src->default_bw;
+	dst->max_bw = src->max_bw;
+	dst->max_rsv_bw = src->max_rsv_bw;
+	dst->admin_grp = src->admin_grp;
+	for (int i = 0; i < MAX_CLASS_TYPE; i++)
+		dst->unrsv_bw[i] = src->unrsv_bw[i];
+	admin_group_copy(&dst->ext_admin_grp, &src->ext_admin_grp);
+	dst->rmt_as = src->rmt_as;
+	memcpy(&dst->rmt_ip, &src->rmt_ip, sizeof(dst->rmt_ip));
+	dst->av_delay = src->av_delay;
+	dst->min_delay = src->min_delay;
+	dst->max_delay = src->max_delay;
+	dst->delay_var = src->delay_var;
+	dst->pkt_loss = src->pkt_loss;
+	dst->res_bw = src->res_bw;
+	dst->ava_bw = src->ava_bw;
+	dst->use_bw = src->use_bw;
+}
+
 struct if_link_params *if_link_params_get(struct interface *ifp)
```

Comment:
I prefer avoiding such functions (see my comments in lib/zclients.c) for performance and safer code

Reply:
changed